epub33/core/index.html

```diff
@@ -5301,11 +5332,22 @@ <h4>File and Directory Structure</h4>
 				</section>
 
 				<section id="sec-container-iri">
-					<h4>Relative URLs for Referencing Other Components</h4>
+					<h4>URLs in the OCF Abstract Container</h4>
 
-					<p>Files within the <a>OCF Abstract Container</a> MUST reference each other via <a
-							href="https://url.spec.whatwg.org/#relative-url-with-fragment-string"
-							>relative-URL-with-fragment strings</a> [[URL]].</p>
+					<p>The <a>container root URL</a> is the <a data-cite="url#concept-url">URL</a> [[URL]] of the
+						<a>Root Directory</a>. It is implementation specific, but MUST abide to the following requirements:</p>
+
+					<ul>
+						<li>The result of <a data-cite="url#concept-url-parser">parsing </a> "<code>/</code>" with the <a>container root URL</a> as <a data-cite="url#concept-base-url">base</a> MUST be the <a>container root URL</a>.</li>
+						<li>The result of <a data-cite="url#concept-url-parser">parsing </a> "<code>..</code>" with the <a>container root URL</a> as <a data-cite="url#concept-base-url">base</a> MUST be the <a>container root URL</a>.</li>	
+					</ul>
+
+					<p>The <a>container URL</a> of a file or directory in the <a>OCF Abstract Container</a> is the result of parsing the file's <a>Path Name</a> with the <a>container root URL</a> as <a data-cite="url#concept-base-url">base</a>.</p>
```

Comment:
"parsing" should be a linked reference to the URL Parser algorithm. Alternatively, this could be made more explicit:

>… is the result of applying the URL Parser to the file's Path Name with…

Reply:
Do we need a note to explain that the parsing algorithm will create validation errors (but still a valid result) if the path contains characters not allowed in URLs?

Reply:
I have just added a reference to the url standard, to be consistent with all the other occurences.

Reply:
As for the extra note: we do not do this at any other places. And, also, we do not define how the errors are handled, so I'd prefer not to get there now (we may want to take a more general look at error handling at a later point). I presume that epubcheck does check the validity of the urls in the package document...

Reply:
Yes, EPUBCheck does check the validity of URLs. This is a conformance statement of the spec, which mandates valid URL strings.

But the issue here is different: we define "Content URL" as the result of a parsing operation that we know by design can legitimately produce validation errors (but not failures).

For instance, if the path of a container entry is: "OPS/My File.xhtml", the parsing will produce a validation error but will still succeed in returning a URL where the space character is encoded as "%20".